### PR TITLE
feature/nav_data_rendering_improvements

### PR DESCRIPTION
**Pull request by TheEmidee**
Fixed and improved debug rendering of the SVO

### CODE REVIEW

File: Source/SVONavigation/Private/SVONavDataRenderingComponent.cpp [hunk 1]
```diff
@@ -1,328 +1,113 @@
...
-#include "Engine/Canvas.h"
+#include <Engine/Canvas.h>
 #endif
 
-#include "Debug/DebugDrawService.h"
-
-#include <Engine/CollisionProfile.h>
-
 #if WITH_EDITOR
 #include "Editor.h"
```
Comment (Frickerson) [inline Source/SVONavigation/Private/SVONavDataRenderingComponent.cpp:14]: <> for both

File: Source/SVONavigation/Private/SVONavDataRenderingComponent.cpp [hunk 2]
```diff
@@ -1,328 +1,113 @@
...
-    if ( DebugInfos.bDebugDrawsOccludedLeaves )
-    {
-        for ( const auto & bounds_data : navigation_bounds )
+        if ( debug_infos.bDebugDrawsLayers )
         {
-            const auto & octree_data = bounds_data.GetOctreeData();
-            const auto occluded_leaf_voxel_size = octree_data.GetLayer( 0 ).GetVoxelHalfSize() * 0.25f;
-            const auto & leaves = octree_data.GetLeaves();
+            const auto corrected_layer_index = FMath::Clamp( static_cast< int >( debug_infos.LayerIndexToDraw ), 0, layer_count - 1 );
+            const auto half_voxel_size = navigation_bounds_data.GetOctreeData().GetLayer( corrected_layer_index ).GetVoxelHalfExtent();
```
Comment (Frickerson) [inline Source/SVONavigation/Private/SVONavDataRenderingComponent.cpp:56]: half_voxel_extent to keep the same naming going

Comment (TheEmidee) [reply Source/SVONavigation/Private/SVONavDataRenderingComponent.cpp:56]: fixed in a later PR

File: Source/SVONavigation/Private/SVONavDataRenderingComponent.cpp (hunk 1 above)
Comment (TheEmidee) [reply Source/SVONavigation/Private/SVONavDataRenderingComponent.cpp:14]: Fixed in a later pR

**Pull request merged**